### PR TITLE
Default compilation for Albert, Distilbert, Roberta MaskedLM

### LINKED ISSUE #830: Add compilation defaults for the MaskedLM task models
In https://github.com/keras-team/keras-nlp/issues/709 we added compilation defaults to all classification tasks. We should also do the same for all `XXMaskedLM` models. E.g. [bert](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/bert/bert_masked_lm.py). Here's the full list of models.

- [ ] `AlbertMaskedLM`
- [ ] `BertMaskedLM`
- [ ] `DebertV3MaskedLM`
- [ ] `DistilBertMaskedLM`
- [ ] `FNetMaskedLM`
- [ ] `RobertaMaskedLM`

https://github.com/keras-team/keras-nlp/pull/714 is a good template PR. We can probably just use Adam and `1e-5` or `2e-5` as a learning rate for now. Though a little experimentation for each model to make sure the task does converge would be helpful. This [colab](https://gist.github.com/mattdangerw/b16c257973762a0b4ab9a34f6a932cc1) may be a helpful starting place.

### PR DESCRIPTION
**Pull request by shivance**
Partially fixes #830

### CODE REVIEW

**Comment by shivance** [comment]
Still working on experimentation with LRs and convergence.

**Comment by mattdangerw** [comment]
@shivance thanks! That was going to be my first question.

Code looks good though!

File: keras_nlp/models/albert/albert_masked_lm.py [hunk 1]
```diff
@@ -142,6 +143,13 @@ def __init__(self, backbone, preprocessor=None, **kwargs):
         self.backbone = backbone
         self.preprocessor = preprocessor
 
+        self.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+            optimizer=keras.optimizers.Adam(5e-5),
+            metrics=keras.metrics.SparseCategoricalAccuracy(),
```
Comment (mattdangerw) [inline keras_nlp/models/albert/albert_masked_lm.py:149]: Actually, I think we want weighted_metrics here as we do pass sampler weights for this task.

Comment (shivance) [reply keras_nlp/models/albert/albert_masked_lm.py:149]: Sure Matt.

**Comment by shivance** [comment]
Hey @mattdangerw , could you please once review keras-io semantic similarity tutorial as well ! Thanks

**Comment by shivance** [comment]
Notebook : https://www.kaggle.com/code/shivanshuman/does-tensorflow-task-converge

## Roberta
Having hard time with roberta, even with batch size of 16, repeatedly getting OOM. Still for three different LR, `lr=5e-5` , outperforms other two.

image

## Albert :
 Something is not quite right here. 
#831 mentioned the same issue regarding albert classifiers.

## DistilBert

**WIP**

**Comment by mattdangerw** [comment]
Thank you! This is super helpful analysis, super appreciate it!

Maybe let's go with 5e-5 everywhere? As performance is almost the same on distilbert. Lower rates are probably "more conservative" in terms of instability. And having the same number will be simpler for now.

**Comment by shivance** [comment]
Sounds good, @mattdangerw, made the change.

**Comment by shivance** [comment]
Also, the run of Albert with AdamW just finished, 

image

AdamW is still in experimental api of tensorflow, so shall I go ahead and put Adam Optimizer for albert?

**Review by mattdangerw — APPROVED** [review]
Great work! Thanks so much for doing all the extra testing behind the scenes here.
